File: antd-lua/examples/06-private-data.lua
--- Example: Store and retrieve private (encrypted) data.

local antd = require("antd")

local client = antd.new_client()

-- Store private data (encrypted on the network)
local secret = "This is my private data"
local result, err = client:data_put_private(secret)
if err then
    print("Put error: " .. err.message)
    os.exit(1)
end

print("Private data stored")
print("Data map: " .. result.address)
print("Cost: " .. result.cost .. " atto")

-- Retrieve private data using the data map
-- The data map is required to decrypt the data — keep it safe!
local retrieved, err2 = client:data_get_private(result.address)
if err2 then
    print("Get error: " .. err2.message)
    os.exit(1)
end

print("Retrieved: " .. retrieved)

File: antd-py/examples/06_private_data.py
"""Example 06: Private (encrypted) data round-trip.

Private data is encrypted before storage. The returned data map
is required to retrieve and decrypt the data.
"""

from antd import AntdClient

client = AntdClient()

# Store private data
secret_message = b"This message is encrypted on the network"
result = client.data_put_private(secret_message)
data_map = result.address  # for private data, address holds the data map
print(f"Data map: {data_map}")
print(f"Cost: {result.cost} atto tokens")

# Retrieve and decrypt
retrieved = client.data_get_private(data_map)
print(f"Decrypted: {retrieved.decode()}")

assert retrieved == secret_message, "Private data round-trip mismatch!"
print("Private data round-trip OK!")
